### ISSUE TITLE
Icon not following template

### ISSUE THREAD

**Issue by Shadewing666** [danieldotnl/ha-multiscrape#171]
## Version of the custom_component

v6.6.0

## Configuration

```yaml
multiscrape:
  - name: Rubbish Scraper
    resource: https://www.aucklandcouncil.govt.nz/rubbish-recycling/rubbish-recycling-collections/Pages/collection-day-detail.aspx?an=12342123382
    scan_interval: 3600
    sensor:
      - unique_id: next_collection
        name: Next Collection
        select: ".links > span:nth-child(3)"
        value_template: >
          {% if value == 'Recycle' %}
            Recycling in next collection
          {% else %}
            Rubish only next collection
          {% endif %}
        icon: >-
          {% if value == 'Recycle' %}
            mdi:recycle
          {% else %}
            mdi:delete
          {% endif %}
      - unique_id: next_collection_date
        name: Next Collection Date
        select: ".m-r-1"
```

## Describe the bug

The icon is not updating to mdi:recycle even though the value is updating to "Recycling in next collection". The 2 templates use the same if statement but the icon template is not updating.

Please note, the address number or "an" in the url has been changed for privacy reasons. This 'an' returns the same result.

## Debug log

This is the debug output after I use the multiscrape.trigger_rubbish_scraper service.

```text

2022-12-28 09:55:42.932 INFO (MainThread) [custom_components.multiscrape] Multiscrape triggered by service: <ServiceCall multiscrape.trigger_rubbish_scraper (c:01GNANX82HFGGCB9ZJZ3HJ1M9V)>
2022-12-28 09:55:42.932 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Rubbish Scraper # New run: start (re)loading data from resource
2022-12-28 09:55:42.932 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Rubbish Scraper # Rendered resource template into: https://www.aucklandcouncil.govt.nz/rubbish-recycling/rubbish-recycling-collections/Pages/collection-day-detail.aspx?an=12342123382
2022-12-28 09:55:42.933 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Rubbish Scraper # Request data from https://www.aucklandcouncil.govt.nz/rubbish-recycling/rubbish-recycling-collections/Pages/collection-day-detail.aspx?an=12342123382
2022-12-28 09:55:42.933 DEBUG (MainThread) [custom_components.multiscrape.http] Rubbish Scraper # Executing page-request with a get to url: https://www.aucklandcouncil.govt.nz/rubbish-recycling/rubbish-recycling-collections/Pages/collection-day-detail.aspx?an=12342123382.
2022-12-28 09:55:43.654 DEBUG (MainThread) [custom_components.multiscrape.http] Rubbish Scraper # Response status code received: 200
2022-12-28 09:55:43.655 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Loading the content in BeautifulSoup.
2022-12-28 09:55:43.708 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Rubbish Scraper # Data succesfully refreshed. Sensors will now start scraping to update.
2022-12-28 09:55:43.708 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Finished fetching multiscrape data in 0.776 seconds (success: True)
2022-12-28 09:55:43.709 DEBUG (MainThread) [custom_components.multiscrape.sensor] Rubbish Scraper # Next Collection # Start scraping to update sensor
2022-12-28 09:55:43.710 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection # Tag selected: <span class="icon-recycle"><span class="sr-only">Recycle</span></span>
2022-12-28 09:55:43.710 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection # Selector result: Recycle
2022-12-28 09:55:43.710 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection # Applying value_template on selector result
2022-12-28 09:55:43.710 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection # Final selector value: Recycling in next collection
2022-12-28 09:55:43.711 DEBUG (MainThread) [custom_components.multiscrape.sensor] Rubbish Scraper # Next Collection # Selected: Recycling in next collection
2022-12-28 09:55:43.711 DEBUG (MainThread) [custom_components.multiscrape.entity] Rubbish Scraper # Next Collection # Icon template rendered and set to: mdi:delete
2022-12-28 09:55:43.711 DEBUG (MainThread) [custom_components.multiscrape.entity] Rubbish Scraper # Next Collection # Sensor updated and state written to HA
2022-12-28 09:55:43.711 DEBUG (MainThread) [custom_components.multiscrape.sensor] Rubbish Scraper # Next Collection Date # Start scraping to update sensor
2022-12-28 09:55:43.713 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection Date # Tag selected: <span class="m-r-1">Saturday 31 December</span>
2022-12-28 09:55:43.713 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection Date # Selector result: Saturday 31 December
2022-12-28 09:55:43.713 DEBUG (MainThread) [custom_components.multiscrape.scraper] Rubbish Scraper # Next Collection Date # Final selector value: Saturday 31 December
2022-12-28 09:55:43.713 DEBUG (MainThread) [custom_components.multiscrape.sensor] Rubbish Scraper # Next Collection Date # Selected: Saturday 31 December
2022-12-28 09:55:43.713 DEBUG (MainThread) [custom_components.multiscrape.entity] Rubbish Scraper # Next Collection Date # Sensor updated and state written to HA


```

**Comment by Shadewing666** [comment]
I made an error in my select and have had to change the if statements as a result. The icon still doesn't change though.
New code:
```
        select: ".card-block > .links:nth-child(2)"
        value_template: >
          {% if value is search("Recycle", ignorecase=True) -%}
            Recycling in next collection
          {%- else -%}
            Rubish only next collection
          {%- endif %}
        icon: >-
          {% if value is search("Recycle", ignorecase=True) -%}
            mdi:recycle
          {%- else -%}
            mdi:delete
          {%- endif %}
```

**Comment by Shadewing666** [comment]
Well, I just figured it out when I looked at that last log. The icon template 'value' is not the result of the select, it is the result of the value_template.

I change the icon code as below and it is now working.

```
        icon: >-
          {% if value == "Recycling in next collection" -%}
            mdi:recycle
          {%- else -%}
            mdi:delete
          {%- endif %}
```
This does not match your README.md which says "Defines the icon or a template for the icon of the sensor. **The value of the _selector_ is provided as input for the template.**" I suggest changing it to "The value of the value_tempate is provided as input for the template." or "The result of the rendered value_tempate is provided as input for the template."
